src/main/java/seedu/address/ui/HelpWindow.java

```diff
@@ -1,102 +1,113 @@
-     * @throws IllegalStateException
-     *     <ul>
-     *         <li>
-     *             if this method is called on a thread other than the JavaFX Application Thread.
-     *         </li>
-     *         <li>
-     *             if this method is called during animation or layout processing.
-     *         </li>
-     *         <li>
-     *             if this method is called on the primary stage.
-     *         </li>
-     *         <li>
-     *             if {@code dialogStage} is already showing.
-     *         </li>
-     *     </ul>
+     * Shows the help window and centers it on screen.
+     * <p>If the window is already visible, this call is a no-op except for bringing
+     * the window to the front due to JavaFX semantics.</p>
      */
     public void show() {
-        logger.fine("Showing help page about the application.");
+        logger.fine("Showing help window.");
         getRoot().show();
         getRoot().centerOnScreen();
     }
 
     /**
-     * Returns true if the help window is currently being shown.
+     * Returns whether the help window is currently visible.
+     *
+     * @return {@code true} if the underlying {@link Stage} is showing; {@code false} otherwise.
      */
     public boolean isShowing() {
         return getRoot().isShowing();
     }
 
     /**
-     * Hides the help window.
+     * Hides the help window without disposing it, allowing it to be shown again later via {@link #show()}.
      */
     public void hide() {
         getRoot().hide();
     }
 
     /**
-     * Focuses on the help window.
+     * Hides the help window without disposing it, allowing it to be shown again later via {@link #show()}.
      */
     public void focus() {
         getRoot().requestFocus();
     }
 
-    /**
-     * Copies the URL to the user guide to the clipboard.
-     */
-    @FXML
-    private void copyUrl() {
-        final Clipboard clipboard = Clipboard.getSystemClipboard();
-        final ClipboardContent url = new ClipboardContent();
-        url.putString(USERGUIDE_URL);
-        clipboard.setContent(url);
+    private void loadUserGuide() {
+        try {
+            String md = readUserGuideMarkdown();
+            if (md == null) {
+                webView.getEngine().loadContent("<h2>UserGuide not found</h2>");
+                return;
+            }
+            Parser parser = Parser.builder().build();
+            HtmlRenderer renderer = HtmlRenderer.builder().build();
+            Node document = parser.parse(md);
+            String html = "<html><head><meta charset='UTF-8'></head><body style='font-family:sans-serif;padding:1em;'>"
+                    + renderer.render(document) + "</body></html>";
+            webView.getEngine().loadContent(html);
+        } catch (Exception e) {
+            webView.getEngine().loadContent("<h2>Failed to load UserGuide:</h2><pre>" + e.getMessage() + "</pre>");
+        }
+    }
+
+    private String readUserGuideMarkdown() throws Exception {
```

Comment:
Can the exception specified be more specific? 


```suggestion
    private String readUserGuideMarkdown() throws IOException {
```